Icebnb/Icebnb/RadiusController.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  RadiusController.swift
+//  Icebnb
+//
+//  Created by Robert Mogos on 12/10/2017.
+//  Copyright © 2017 Robert Mogos. All rights reserved.
+//
+
+import UIKit
+import KUIPopOver
+import InstantSearchCore
+import AlgoliaSearch
+
+class RadiusController: UIViewController {
+  @IBOutlet weak var slider: PRGRoundSlider!
+  let searcher: Searcher
+  
+  init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?, searcher: Searcher) {
+    self.searcher = searcher
+    super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
+  }
+  
+  required init?(coder aDecoder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+  
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    
+    var existingValue = Geo.defaultRadius
+    switch searcher.params.aroundRadius {
+      case let .explicit(value)?:
+        existingValue = Double(value)
+        break;
+      default:
+        break;
+    }
+    slider.value = CGFloat((existingValue - Geo.startRadius) / (Geo.endRadius - Geo.startRadius))
+    slider.messageForValue = { [weak searcher] value in
+      let meters = round(Geo.startRadius + Double(value) * (Geo.endRadius - Geo.startRadius))
+      searcher?.params.aroundRadius = Query.AroundRadius.explicit(UInt(meters))
```

Comment:
remove the "?" after force unwrapping

Reply:
I passed a weak reference to the searcher in the closure, hence the ?.